devicesController.go

```diff
@@ -2,16 +2,18 @@ package main
 
 import (
 	"encoding/json"
+	"github.com/gorilla/mux"
 	"net/http"
 	"strconv"
 )
 
-type DeviceCreator interface {
+type IDeviceService interface {
```

Comment:
`Isomething` is not really a golang way of naming :) I know you have a name clash here and I don't have an idea for better name so if you find one use it, otherwise ignore me :D

Reply:
You guessed my intentions.
In this particular case, the interface could be private and this approach could prevent name clash. In golang, interface should be define as public by default, or only when necessary?

Reply:
I would say that it depends on the use case. When you create the interface just to be able to mock in tests I think that private interface will be fine. But if you create the interface so that someone can provide different implementation it has to be public.

In this case I think that private interface in this case private interface would do just fine - implementation is provided in some factory or main so there is not too much of moving parts there.